chronos/transports/datagram.nim

```diff
@@ -28,7 +30,9 @@ type
     writer: Future[void]        # Writer vector completion Future
 
   DatagramCallback* = proc(transp: DatagramTransport,
-                           remote: TransportAddress): Future[void] {.gcsafe.}
+                           message: seq[byte],
+                           remote: TransportAddress,
+                           error: OSErrorCode): Future[void] {.gcsafe.}
```

Comment:
Seems like `error` which can happen here is either not realistic or unrelated to remote-specific message receiving. Thus I would consider removing the `error` from this function signature. For remote-unrelated events I would introduce another callback like `onNetworkChanged`.

Reply:
Sorry, but `chronos` is cross-platform framework, and if Linux has small number of errors, BSD/MacOS and Windows has much more, so i wish to leave error code here.